test/unit/lib/ImagerySource.test.js

```diff
@@ -111,6 +153,216 @@ describe('ImagerySource', () => {
       assert.equal(source2.isLocatorOverlay(), true);
     });
   });
+
+
+  describe('_localeDateString', () => {
+    it('returns null if the input is not provided', () => {
+      const source = new Rapid.ImagerySource(context, { id: 'test' });
+      assert.strictEqual(source._localeDateString(), null);
+    });
+
+    it('returns null if the input is not a valid date', () => {
+      const source = new Rapid.ImagerySource(context, { id: 'test' });
+      assert.strictEqual(source._localeDateString('not a date'), null);
+    });
+
+    it('returns a formatted date string if the input is a valid date', () => {
+      const source = new Rapid.ImagerySource(context, { id: 'test' });
+      assert.strictEqual(source._localeDateString('2020-01-01'), '2020-01-01');
+    });
+  });
+
+
+  describe('_vintageRange', () => {
+    it('returns undefined if the input does not have a start or end', () => {
+      const source = new Rapid.ImagerySource(context, { id: 'test' });
+      assert.strictEqual(source._vintageRange({}), undefined);
+    });
+
+    it('returns a string with the start date if only the start is provided', () => {
+      const source = new Rapid.ImagerySource(context, { id: 'test' });
+      assert.strictEqual(source._vintageRange({ start: 'Jan 1, 2020' }), 'Jan 1, 2020 - ?');
+    });
+
+    it('returns a string with the end date if only the end is provided', () => {
+      const source = new Rapid.ImagerySource(context, { id: 'test' });
+      assert.strictEqual(source._vintageRange({ end: 'Dec 31, 2020' }), '? - Dec 31, 2020');
+    });
+
+    it('returns a range string if both the start and end are provided', () => {
+      const source = new Rapid.ImagerySource(context, { id: 'test' });
+      assert.strictEqual(source._vintageRange({ start: 'Jan 1, 2020', end: 'Dec 31, 2020' }), 'Jan 1, 2020 - Dec 31, 2020');
+    });
+  });
+
+
+  describe('fetchTilemap', () => {
+    // Save the original fetch function
+    const originalFetch = global.fetch;
+    afterEach(() => {
+      // Restore the original fetch function after each test
+      global.fetch = originalFetch;
+    });
+    it('fetches a tilemap and updates the zoom extent', async () => {
+      const source = new Rapid.ImagerySourceEsri(context, {
+        id: 'anyid',
+        template: 'http://example.com/tile/{z}/{x}/{y}?blankTile=false'
+      });
+      // Mock the fetch function to return a tilemap with all tiles present
+      global.fetch = () => Promise.resolve({
```

Comment:
We should probably use `fetchMock` like other service tests do, instead of changing `global.fetch`:

https://github.com/facebook/Rapid/blob/a665ef74de559f222bf3bb397ac06cdcecc71c96/test/browser/services/OsmService.test.js#L171-L175